test/passes/remove-unused-names_merge-blocks_all-features.txt

```diff
@@ -166,14 +166,11 @@
     (i32.const 20)
    )
   )
-  (return
-   (block
-    (drop
-     (i32.const 10)
-    )
-    (unreachable)
-   )
+  (drop
+   (i32.const 10)
   )
+  (unreachable)
+  (return)
```

Comment:
This looks like a loss of the ability to test such a nested block in a return - is this fixable?

Reply:
The surrounding function has no return values, so the `return` has no business having children. In the old parser we allowed this anyway if the children happened to be unreachable, but I actually think the new behavior makes more sense. Perhaps we should add a check to our validator that only returns in functions that have return values can have children.

Reply:
Thanks, I missed it had no returns from the function. This does sound better then, I agree. Validator improvement makes sense too.